src/EditorFeatures/Test/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs

```diff
@@ -102,17 +94,103 @@ public TestParameters WithIncludeDiagnosticsOutsideSelection(bool includeDiagnos
         protected TestWorkspace CreateWorkspaceFromOptions(
             string initialMarkup, TestParameters parameters)
         {
-            var workspace = TestWorkspace.IsWorkspaceElement(initialMarkup)
-                 ? TestWorkspace.Create(initialMarkup, openDocuments: false)
-                 : CreateWorkspaceFromFile(initialMarkup, parameters);
+            // For CodeStyle layer testing, we create an .editorconfig at project root
+            // to apply the options as workspace options are not available in CodeStyle layer.
+            // Otherwise, we apply the options directly to the workspace.
```

Comment:
❔ Why add all of this complexity? The code style layer already supports testing with .editorconfig, since IDE0055 requires it.

Reply:
@sharwell This is the only test framework file that needed changes to allow AnalyzerConfigOptions based testing in CodeStyle layer. We can revert all this code once all our IDE analyzer/code fix tests are moved to Microsoft.CodeAnalysis.Testing.